File: rust/src/p0036.rs
/*
Project Euler Problem 36

Rust implementing a `reverse_bits()` function made this much more pleasant than expected

Problem:

The decimal number, 585 = 10010010012 (binary), is palindromic in both bases.

Find the sum of all numbers, less than one million, which are palindromic in
base 10 and base 2.

(Please note that the palindromic number, in either base, may not include
leading zeros.)
*/
use crate::include::utils::{is_palindrome,Answer};

pub fn p0036() -> Answer {
    let mut answer: u64 = 0;
    for x in 1..1000000u64 {
        if is_palindrome(format!("{x:b}")) && is_palindrome(x) {
            answer += x;
        }
    }
    return Answer::Int(answer.into());
}

File: python/src/p0036.py
"""
Project Euler Problem 36

This one ended up being very easy thanks to the datetime library

Problem:

The decimal number, 585 = 10010010012 (binary), is palindromic in both bases.

Find the sum of all numbers, less than one million, which are palindromic in
base 10 and base 2.

(Please note that the palindromic number, in either base, may not include
leading zeros.)

"""
from .lib.utils import is_palindrome


def bin_repr(x: int) -> str:
    return bin(x)[2:]


def main() -> int:
    answer = 0
    for x in range(1, 1000000):
        if is_palindrome(x) and is_palindrome(x, rep_func=bin_repr):
            answer += x
    return answer
